lib/dialects/postgres/index.js

```diff
@@ -273,22 +272,20 @@ Object.assign(Client_PG.prototype, {
   },
 
   canCancelQuery: true,
-  cancelQuery(connectionToKill) {
-    const acquiringConn = this.acquireConnection();
-
+  async cancelQuery(connectionToKill) {
     // Error out if we can't acquire connection in time.
     // Purposely not putting timeout on `pg_cancel_backend` execution because erroring
     // early there would release the `connectionToKill` back to the pool with
     // a `KILL QUERY` command yet to finish.
-    return acquiringConn.then((conn) => {
-      return this._wrappedCancelQueryCall(conn, connectionToKill).finally(
-        () => {
-          // NOT returning this promise because we want to release the connection
-          // in a non-blocking fashion
-          this.releaseConnection(conn);
-        }
-      );
-    });
+    const conn = await this.acquireConnection();
+
+    try {
+      return await this._wrappedCancelQueryCall(conn, connectionToKill);
+    } finally {
+      // NOT returning this promise because we want to release the connection
+      // in a non-blocking fashion
+      this.releaseConnection(conn);
+    }
```

Comment:
Logic appears to be equivalent to the logic that it replaced.

(We should probably consider adding an `await` to the `this.releaseConnection(..)` call in a future PR, though.  Well, more specifically: figure out a good way to handle unhandled promise rejections from that call)